sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs

```diff
@@ -291,6 +291,8 @@ public PipelineResponse(string requestId, HttpResponseMessage responseMessage)
 
             public override int Status => (int)_responseMessage.StatusCode;
 
+            public override string ReasonPhrase => _responseMessage.ReasonPhrase;
```

Comment:
I'm not sure how much we want to sanitize error responses and exceptions.

Reply:
In this particular case, you're not including headers or anything that it would be reasonable to assert strong rules against, so that means we're implicitly trusting the service to do the right ting.   Personally, I'm not overly comfortable with a blanket statement that all services are screening privileged data from their response.   

I don't see much of an action that we can take given that we're transporting an opaque value, but I would strongly advise that we make that clear in our contract.   I'd recommend that we provide XML comment docs that clearly state that we make no guarantee that this particular exception is safe to log, as the value is controlled by an external dependency.

Reply:
> I don't see much of an action that we can take given that we're transporting an opaque value, but I would strongly advise that we make that clear in our contract. I'd recommend that we provide XML comment docs that clearly state that we make no guarantee that this particular exception is safe to log, as the value is controlled by an external dependency.

I think no one would see it. Usually exceptions are logged via `try { ... } catch (Exception ex) { Logger.Log(ex); }` rarely people read the doc comments to exceptions.

Reply:
There is a difference between "nobody will read it" and "we're protecting our liability by making responsibilities clear."     Whether it is read or not, it's easy to refer to as "we told you and you did it anyway" versus "why didn't you go read the source and try to understand the call chain."

I can see this particular exception as something that gets exposed up the stack to SDK consumers and not something that is only used internally.